surf/apps/communities/views.py

```diff
@@ -161,21 +155,54 @@ def _delete_collections(instance, collections):
         collections = Collection.objects.filter(id__in=collections).all()
         instance.collections.remove(*collections)
 
-    @staticmethod
-    def _check_access(user, instance=None, collection_ids=None):
-        """
-        Check if user is active and admin of community
-        :param user: user
-        :param instance: community instance
-        :param collection_ids: list of identifiers of collections
-        added/deleted to/from community
-        """
-        if not user or not user.is_authenticated:
-            raise AuthenticationFailed()
+
+class CommunityDetailAPIView(APIView):
+    def get(self, request, *args, **kwargs):
+        language_code = kwargs['language_code'].upper()
+        community_id = kwargs['community_id']
+
+        try:
+            detail_object = CommunityDetail.objects.get(community__id=community_id, language_code=language_code)
+        except ObjectDoesNotExist:
+            return Response(f"Community has no details for this language.")
+
+        return Response(CommunityDetailSerializer(detail_object).data)
+
+    def post(self, request, *args, **kwargs):
+        language_code = kwargs['language_code'].upper()
+        community_id = kwargs['community_id']
+        community_object = Community.objects.get(id=community_id)
+        try:
+            check_access_to_community(request.user, community_object)
+        except AuthenticationFailed as exc:
+            return Response(str(exc), status=401)
+        detail_object, created = CommunityDetail.objects.get_or_create(community=community_object,
+                                                                       language_code=language_code)
         try:
-            Team.objects.get(community=instance, user=user)
-        except ObjectDoesNotExist as exc:
-            raise AuthenticationFailed(f"User {user} is not a member of community {instance}. Error: \"{exc}\"")
-        except MultipleObjectsReturned as exc:
-            # if somehow there are user duplicates on a community, don't crash
-            logger.warning(f"User {user} is in community {instance} multiple times. Error: \"{exc}\"")
+            for attr, value in request.data.items():
+                if value is not None:
+                    setattr(detail_object, attr, value)
+            detail_object.clean()
+            detail_object.save()
+        except Exception as exc:
+            return Response(f"Error while updating the requested object: {str(exc)}.", status=400)
+
+        return Response(CommunityDetailSerializer(detail_object).data)
+
+
+def check_access_to_community(user, instance=None):
```

Comment:
Overwriting the GenericAPIView get_object method would allow to insert the usage of this function. The get_object method gets called when DRF tries to retrieve an instance using the parameters from the request.

Reply:
I've done that now, I'm not sure if it's done 100% like it should. I took inspiration from the DRF Permissions page.